src/Castle.Core/DynamicProxy/CustomAttributeInfo.cs

```diff
@@ -169,28 +170,44 @@ private static Expression UnwrapBody(Expression body)
 
 		private static object GetAttributeArgumentValue(Expression arg, bool allowArray)
 		{
-			var constant = arg as ConstantExpression;
-			if (constant == null)
+			if (arg is ConstantExpression constant)
```

Comment:
I noticed that you are repeatedly checking `arg`'s type. That's fine and it enables the nice `is T x` syntax, but if you're willing to micro-optimize, you might get a wee bit better performance by querying & caching `arg.NodeType` just once, then compare the various `ExpressionType` enum values against that (e.g. using a `switch` block).

Reply:
I pushed a fixup commit, let me know if you like it better